### ISSUE TITLE
node cleanup fails when instance doesn't exist any more

### ISSUE THREAD

**Issue by universam1** [aws/karpenter#2100]
## Version

**Karpenter**: v0.13.2

**Kubernetes**: v1.21.12-eks-a64ea69 

## Expected Behavior
Karpenter detects that instance is not existent in AWS, and skips the terminate action.

**Problems found:**
1. NodeTerminationHandler is not possible to be used: https://github.com/aws/aws-node-termination-handler/issues/654
2. When node is gone, the **tag filter is not matching**:  https://karpenter.sh/v0.13.2/getting-started/migrating-from-cas/
```json
{
            "Action": "ec2:TerminateInstances",
            "Condition": {
                "StringLike": {
                    "ec2:ResourceTag/Name": "*karpenter*"
```
3. Karpenter tries to delete a node that is not existent and fails.

## Actual Behavior
When the node is terminated in AWS due to spot interruption and K. was not aware of it, the node is not deleted in Kubernetes and Karpenter loops into this error:
```
terminating instance ip-172-16-101-161.ec2.internal, UnauthorizedOperation: You are not authorized to perform this operation
```

## Steps to Reproduce the Problem
- scale K. to zero instance
- terminate the node, wait until gone
- scale K. up

## Resource Specs and Logs

```json
{
"level": "ERROR",
"time": "2022-07-11T11:45:47.555Z",
"logger": "controller.controller.termination",
"message": "Reconciler error",
"commit": "1f7a67b",
"reconciler group": "",
"reconciler kind": "Node",
"name": "ip-172-16-101-161.ec2.internal",
"namespace": "",
"error": "terminating node ip-172-16-101-161.ec2.internal, terminating cloudprovider instance, terminating instance ip-172-16-101-161.ec2.internal, UnauthorizedOperation: You are not authorized to perform this operation. Encoded authorization failure message: xxxxx\n\tstatus code: 403, request id: xxxx"
}
```

decoded failure message:
```json
{
    "allowed": false,
    "explicitDeny": false,
    "matchedStatements": {
        "items": []
    },
    "failures": {
        "items": []
    },
    "context": {
        "principal": {
            "id": "xxx:be41e23e-xxxx-KarpenterControllerIamRo-6HZ0",
            "arn": "arn:aws:sts::xxxx:assumed-role/xxxx-KarpenterControllerIamRo-6HZ0EFA4MZ0X/be41e23e-xxxx-KarpenterControllerIamRo-6HZ0"
        },
        "action": "ec2:TerminateInstances",
        "resource": "*",
        "conditions": {
            "items": [
                {
                    "key": "aws:Region",
                    "values": {
                        "items": [
                            {
                                "value": "us-east-1"
                            }
                        ]
                    }
                },
                {
                    "key": "aws:Service",
                    "values": {
                        "items": [
                            {
                                "value": "ec2"
                            }
                        ]
                    }
                },
                {
                    "key": "aws:Account",
                    "values": {
                        "items": [
                            {
                                "value": "xxxx"
                            }
                        ]
                    }
                },
                {
                    "key": "aws:ARN",
                    "values": {
                        "items": [
                            {
                                "value": "*"
                            }
                        ]
                    }
                }
            ]
        }
    }
}
```

**Comment by bwagner5** [comment]
I tried replicating this today, but things are working cleanly for me within K8s and Karpenter.  I think my next step is to test this will just EC2. 

1. Create IAM Role w/ a TerminateInstances & DescribeInstances policy w/ tag condition 
2. Create instance with that tag
3. Terminate the instance while assuming the role from step 1 (simulate a spot interruption or system failure)
4. Terminate the instance after it is terminated (should see permission error)
5. DescribeInstance after it is terminated to see if it is describable after termination


IMO here are the two remidiations:

1. If we are able to Describe the instance after termination, then we need to add a check to see if we get an authorization error, and then try to describe the instance. If we get another authorization error, then just log it. 

2. If we are unable to Describe the instance after termination, then catch the authorization error and log it, but don't return an error.

**Comment by bwagner5** [comment]
It seems like something else might be going wrong here... I just did the test outlined above. I realized that you can't set a condition on DescribeInstances, but even after I terminated the instance with the conditional IAM policy, I'm able to keep terminating the instance without a permissions issue.  I'll keep trying to see if maybe after 1 hour, when the instance is reaped from EC2, that a permissions error is thrown on a terminate. But for a spot instance interruption, Karpenter should realize the node is gone well before the instance is cleaned up.

**Comment by bwagner5** [comment]
Ok, after about an hour and a half, I was able to get the Unauthorized message in response to a TerminateInstances request.

**Comment by bwagner5** [comment]
The path forward seems to be to add a Describe call on a authz error for termination. If we get another auth error, then keep failing in a loop like you experienced. If we see the instance is terminated, then don't log.

**Comment by dewjam** [comment]
I believe this issue is also related.  https://github.com/aws/karpenter/issues/1796

**Comment by dewjam** [comment]
This should be fixed by #2253 .
